src/ssh/cockpitsshrelay.c

```diff
@@ -922,13 +921,18 @@ do_key_identity_auth (CockpitSshData *data)
   g_autofree gchar *pub_key_path = NULL;
   const gchar *msg;
 
+  gchar *default_identities[] = { "id_dsa", "id_ecdsa", "id_ecdsa_sk", "id_ed25519", "id_ed25519_sk", "id_rsa", NULL };
+  int next_identity = 0;
+
   rc = ssh_options_get (data->session, SSH_OPTIONS_IDENTITY, &identity);
   if (rc != SSH_OK)
     {
       g_debug ("Unable to get identity from config");
       return rc;
     }
 
+ again:
```

Comment:
Ugh -- any chance you could write this as a for loop instead of goto? It's hard to see what happens to e. g. the static struct assignment (`struct CockpitSshPromptData pd = { data, identity };`) across such jumps.

Reply:
Done in #14551